Makefile

```diff
@@ -274,148 +275,283 @@ docker-ci-integration:
 		-w "/go/flow" "$(CONTAINER_REGISTRY)/golang-cmake:v0.0.7" \
 		make ci-integration
 
-.PHONY: docker-build-collection
-docker-build-collection:
-	docker build -f cmd/Dockerfile  --build-arg TARGET=./cmd/collection --build-arg COMMIT=$(COMMIT)  --build-arg VERSION=$(IMAGE_TAG) --build-arg GOARCH=$(GOARCH) --build-arg CGO_FLAG=$(CRYPTO_FLAG) --target production \
+# only works on linux
+.PHONY: install-cross-build-tools
+install-cross-build-tools:
+	apt-get update
+	apt-get -y install apt-utils gcc-aarch64-linux-gnu
+
+.PHONY: docker-native-build-collection
+docker-native-build-collection:
+	docker build -f cmd/Dockerfile --build-arg TARGET=./cmd/collection --build-arg COMMIT=$(COMMIT)  --build-arg VERSION=$(IMAGE_TAG) --build-arg GOARCH=$(GOARCH) --build-arg CGO_FLAG=$(CRYPTO_FLAG) --target production \
 		--secret id=git_creds,env=GITHUB_CREDS --build-arg GOPRIVATE=$(GOPRIVATE) \
 		--label "git_commit=${COMMIT}" --label "git_tag=${IMAGE_TAG}" \
 		-t "$(CONTAINER_REGISTRY)/collection:latest" -t "$(CONTAINER_REGISTRY)/collection:$(SHORT_COMMIT)" -t "$(CONTAINER_REGISTRY)/collection:$(IMAGE_TAG)" .
 
-.PHONY: docker-build-collection-without-netgo
-docker-build-collection-without-netgo:
-	docker build -f cmd/Dockerfile --build-arg TARGET=./cmd/collection --build-arg COMMIT=$(COMMIT)  --build-arg VERSION=$(IMAGE_TAG_NO_NETGO) --build-arg GOARCH=$(GOARCH) --build-arg CGO_FLAG=$(CRYPTO_FLAG) --target production \
+.PHONY: docker-build-collection-with-adx
+docker-build-collection-with-adx:
```

Comment:
Optimal case (with ADX and with netgo) - this is called in CD.

Note that the image has the same "tag" as the native builds. Native builds (locally or CI) are not run in CD.

Reply:
so the difference between this an `docker-native-build-collection` is that native allows overriding ADX while this doesn't?

Reply:
This target hardcodes the arguments for a specific target (ideally I would want to hardcode `GOARCH` too, but left it in case we want to call it on `ARM` machines too). 

`docker-native-build-...` set the arguments based on the host (detect ADX support, set `ARCH`)

Reply:
For better clarity and consistency, I hardcoded `amd64` for production targets that are not natively built and not meant for `arm` https://github.com/onflow/flow-go/pull/5254/commits/d3b99bd42237df016693a1be7aecf2245ab9c744 (though I can revert it if needed)